<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>900</width>
    <height>600</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <property name="dockNestingEnabled">
   <bool>false</bool>
  </property>
  <property name="unifiedTitleAndToolBarOnMac">
   <bool>true</bool>
  </property>
  <widget class="QWidget" name="centralWidget">
   <layout class="QHBoxLayout" name="horizontalLayout">
    <property name="spacing">
     <number>0</number>
    </property>
    <property name="leftMargin">
     <number>0</number>
    </property>
    <property name="topMargin">
     <number>0</number>
    </property>
    <property name="rightMargin">
     <number>0</number>
    </property>
    <property name="bottomMargin">
     <number>0</number>
    </property>
    <item>
     <widget class="QWidget" name="widget" native="true"/>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>900</width>
     <height>22</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuFile">
    <property name="title">
     <string>File</string>
    </property>
    <addaction name="actionNew"/>
    <addaction name="actionReload"/>
    <addaction name="separator"/>
    <addaction name="actionExit"/>
   </widget>
   <widget class="QMenu" name="menuWindows">
    <property name="title">
     <string>Windows</string>
    </property>
    <addaction name="actionRegisters"/>
    <addaction name="actionMemory"/>
    <addaction name="actionProgram_memory"/>
    <addaction name="actionCache"/>
    <addaction name="actionCache_statistics"/>
   </widget>
   <widget class="QMenu" name="menuMachine">
    <property name="title">
     <string>Machine</string>
    </property>
    <addaction name="actionRun"/>
    <addaction name="actionPause"/>
    <addaction name="actionStep"/>
    <addaction name="separator"/>
    <addaction name="ips1"/>
    <addaction name="ips5"/>
    <addaction name="ips10"/>
    <addaction name="ipsUnlimited"/>
    <addaction name="separator"/>
    <addaction name="actionRestart"/>
   </widget>
   <addaction name="menuFile"/>
   <addaction name="menuMachine"/>
   <addaction name="menuWindows"/>
  </widget>
  <widget class="QStatusBar" name="statusBar"/>
  <widget class="QToolBar" name="toolBar">
   <property name="windowTitle">
    <string>toolBar</string>
   </property>
   <property name="movable">
    <bool>false</bool>
   </property>
   <property name="allowedAreas">
    <set>Qt::TopToolBarArea</set>
   </property>
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
   <addaction name="actionNew"/>
   <addaction name="actionReload"/>
   <addaction name="separator"/>
   <addaction name="actionRun"/>
   <addaction name="actionPause"/>
   <addaction name="actionStep"/>
   <addaction name="separator"/>
   <addaction name="ips1"/>
   <addaction name="ips5"/>
   <addaction name="ips10"/>
  </widget>
  <action name="actionNew">
   <property name="icon">
    <iconset resource="icons.qrc">
     <normaloff>:/icons/document-import.png</normaloff>:/icons/document-import.png</iconset>
   </property>
   <property name="text">
    <string>New simulation...</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+O</string>
   </property>
  </action>
  <action name="actionExit">
   <property name="icon">
    <iconset resource="icons.qrc">
     <normaloff>:/icons/application-exit.png</normaloff>:/icons/application-exit.png</iconset>
   </property>
   <property name="text">
    <string>Exit</string>
   </property>
   <property name="toolTip">
    <string>Exit program</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+Q</string>
   </property>
  </action>
  <action name="actionRestart">
   <property name="text">
    <string>Restart</string>
   </property>
  </action>
  <action name="actionRun">
   <property name="icon">
    <iconset resource="icons.qrc">
     <normaloff>:/icons/play.png</normaloff>:/icons/play.png</iconset>
   </property>
   <property name="text">
    <string>Run</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+R</string>
   </property>
  </action>
  <action name="actionStep">
   <property name="icon">
    <iconset resource="icons.qrc">
     <normaloff>:/icons/next.png</normaloff>:/icons/next.png</iconset>
   </property>
   <property name="text">
    <string>Step</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+S</string>
   </property>
  </action>
  <action name="actionPause">
   <property name="icon">
    <iconset resource="icons.qrc">
     <normaloff>:/icons/pause.png</normaloff>:/icons/pause.png</iconset>
   </property>
   <property name="text">
    <string>Pause</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+P</string>
   </property>
  </action>
  <action name="ips1">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>1 instruction per second</string>
   </property>
   <property name="iconText">
    <string>1x</string>
   </property>
   <property name="toolTip">
    <string>Run CPU in speed of single instruction per second</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+1</string>
   </property>
  </action>
  <action name="ips5">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>5 instructions per second</string>
   </property>
   <property name="iconText">
    <string>5x</string>
   </property>
   <property name="toolTip">
    <string>Run CPU in speed of 5 instructions per second</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+5</string>
   </property>
  </action>
  <action name="ips10">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>10 instructions per second</string>
   </property>
   <property name="iconText">
    <string>10x</string>
   </property>
   <property name="toolTip">
    <string>Run CPU in speed of 10 instructions per second</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+0</string>
   </property>
  </action>
  <action name="ipsUnlimited">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Unlimited</string>
   </property>
   <property name="toolTip">
    <string>Run CPU without any clock constrains</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+U</string>
   </property>
  </action>
  <action name="actionMemory">
   <property name="text">
    <string>Memory</string>
   </property>
   <property name="toolTip">
    <string>Data memory view</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+M</string>
   </property>
  </action>
  <action name="actionProgram_memory">
   <property name="text">
    <string>Program</string>
   </property>
   <property name="toolTip">
    <string>Program memory view</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+P</string>
   </property>
  </action>
  <action name="actionCache">
   <property name="text">
    <string>Cache content</string>
   </property>
  </action>
  <action name="actionCache_statistics">
   <property name="text">
    <string>Cache statistics</string>
   </property>
  </action>
  <action name="actionRegisters">
   <property name="text">
    <string>Registers</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+D</string>
   </property>
  </action>
  <action name="actionReload">
   <property name="icon">
    <iconset resource="icons.qrc">
     <normaloff>:/icons/reload.png</normaloff>:/icons/reload.png</iconset>
   </property>
   <property name="text">
    <string>Reload simulation</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+Shift+R</string>
   </property>
  </action>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <resources>
  <include location="icons.qrc"/>
 </resources>
 <connections/>
</ui>
